pkg/kubectl/resource/builder.go

```diff
@@ -402,9 +419,9 @@ func (b *Builder) ResourceTypeOrNameArgs(allowEmptySelector bool, args ...string
 		b.ResourceTypes(SplitResourceArgument(args[0])...)
 	case len(args) == 1:
 		b.ResourceTypes(SplitResourceArgument(args[0])...)
-		if b.selector == nil && allowEmptySelector {
+		if b.labelSelector == nil && allowEmptySelector {
```

Comment:
What if fieldSelector is not nil?  What happens then?

Reply:
> What if fieldSelector is not nil? What happens then?

This does not affect `fieldSelector`. Will work as expected.